providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -186,7 +186,14 @@ private <T> ProviderEvaluation<T> resolve(Class<T> type, String key,
             log.debug(message);
             throw new TypeMismatchError(message);
         }
-
+        // if this is an integer and we are trying to resolve a double, convert
+        if (value instanceof Integer && type == Double.class) {
+            value = ((Integer) value).doubleValue();
+        
+        // if this is a doulbe and we are trying to resolve an integer, convert
+        } else if (value instanceof Double && type == Integer.class) {
+            value = ((Double) value).intValue();
+        }
```

Comment:
@Kavindu-Dodan I'm not sure I can think of a more elegant way to do this. At first I though we could just loosen the check on line 197, but since these are boxed types, the cast to `T` at the end isn't enough; we actually have to do a conversion.